### PR TITLE
fix(ReduxQuerySync): Register the sync as soon as possible

### PR DESCRIPTION
**Pull request by grafakus**
Caused by https://github.com/grafana/pyroscope/pull/2692

This PR fixes a race condition where the apps list is loaded before parsing the values of the `from` and `until` URL query parameters. The effect is that the incorrect app/no app is selected in the dropdown.

|Before | After |
|---|---|
|Screenshot 2023-11-17 at 21 13 01|Screenshot 2023-11-17 at 21 09 16|

### CODE REVIEW

File: public/app/redux/useReduxQuerySync.ts [hunk 1]
```diff
@@ -99,6 +99,8 @@ function setupReduxQuerySync() {
   });
 }
 
+const unsubscribe = setupReduxQuerySync();
```
Comment (darrenjaneczek) [inline public/app/redux/useReduxQuerySync.ts:102]: Won't this cause it to be set up all the time?

File: public/app/redux/useReduxQuerySync.ts [hunk 2]
```diff
@@ -99,6 +99,8 @@ function setupReduxQuerySync() {
...
+
 export function useReduxQuerySync() {
-  useEffect(setupReduxQuerySync, []);
+  useEffect(() => unsubscribe, []);
```
Comment (darrenjaneczek) [inline public/app/redux/useReduxQuerySync.ts:105]: Won't this cause it to be unsubscribed on the first unmount, but never `setupReduxQuerySync` again?

Comment (grafakus) [reply public/app/redux/useReduxQuerySync.ts:105]: In the case of the app, it's never unmounted right?

File: public/app/app.tsx [hunk 3]
```diff
@@ -15,15 +15,15 @@ import '@szhsin/react-menu/dist/index.css';
...
-import { useReduxQuerySync } from './redux/useReduxQuerySync';
+import { setupReduxQuerySync } from './redux/useReduxQuerySync';
 import './sass/profile.scss';
 
 const container = document.getElementById('reactRoot') as HTMLElement;
 const root = ReactDOM.createRoot(container);
 
-function App() {
-  useReduxQuerySync();
+setupReduxQuerySync();
```
Comment (darrenjaneczek) [inline public/app/app.tsx:24]: I really don't like the idea of making this a global effect again. Let's explore a better way to make it occur earlier without

Comment (darrenjaneczek) [reply public/app/app.tsx:24]: We will explore it later.

**Review by darrenjaneczek — APPROVED** [review]
Let's do it.

**Pull request merged**